src/mscorlib/src/System/Collections/Generic/Dictionary.cs

```diff
@@ -515,12 +507,26 @@ private struct Entry {
             return false;
         }
 
-        public bool TryGetValue(TKey key, out TValue value) {
-            int i = FindEntry(key);
-            if (i >= 0) {
-                value = entries[i].value;
-                return true;
+        public bool TryGetValue(TKey key, out TValue value)
+        {
+            if (key == null)
```

Comment:
Should not `key == null` be optimized away anyway? I believe so. If not this is something the JIT could/should be taught.